### DISCUSSION TITLE
Meeting: Celo Wallet and (d)App Council #22 [April] 💬

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/celo-monorepo#10248] [Wallet and (d)App Council]
# Monthly Celo Wallet and (d)App Council

##  📆 Quick links 

- When: **Wednesday, April 12 (9am PT/6pm CET)** 
  _I'm OOO, so Subha kindly agreed to host this session!_  🙏 
- Where: **[Google Meet](https://meet.google.com/uwj-wfse-uzm)**
- Agenda: **comments**👇
- Previous: https://github.com/celo-org/celo-monorepo/discussions/10157
- Next: https://github.com/celo-org/celo-monorepo/discussions/10308
- Recording: after meeting

## 🤷‍♂️ What is this?

A monthly meeting for Celo **developers, founders,** **product** people (anyone really!) to come together and
- find consensus (coordinate),
- learn about needs, pain points, and desires
- present and collect technical feedback

## 🗒️  Agenda

- Comment with a short sentence describing what you'd like to discuss
- Don't worry, it can be small or big! 
- Every builder is welcome (and we are very easy going, I promise 😄)
- Upvote topics you're interested in

## Housekeeping

_You can safely ignore this!_

- [x] Send update calendar invite with Github link
- [ ] Encourage additions to the agenda
- [ ] Publicise on forum and Discord
- [ ] RECORD meeting 🎥 
- [ ] Share recording below and on forum
- [x] Create next meeting (link to next meeting above)

**Comment by arthurgousset** [comment]
Reminder: [Please upgrade to WalletConnect v2](https://forum.celo.org/t/walletconnect-v1-to-v2-migration/5184)

If you don't upgrade by Wed, Jun 28 2023, your application might stop working if you are currently using WalletConnect v1.

**Comment by arthurgousset** [comment]
## Admin update

I'm OOO, so Subha kindly agreed to host this session! 🙏
Please remember to record the session for future reference (if possible).
Thank you!

**Comment by aaronmgdr** [comment]
# wallet connect v2

Subha Wallet Connect v2 adoptions: any thoughts? what is the status? 

Jean: Dapps dont seem to have upgraded yet. (bassed on valora analytics) 

Subha: is valora asking dapps to upgrade?

Jean: Would definitely be better for users if dapps upgrade before WV1 termination date

Frederic: (Dapp Dev): what is benefit to upgrade? could there be a backup server for v1?  might do one ourselves if we dont think v2 is worth it. Probably not worth it for us. but maybe for celo in general having a v1 server would be good. 

Eric: (Wallet Dev) one thing that is beneficial of v2 is push notifications.  Most wallets will retire support for v1. (likely)   

Jean: Valora can keep v1, but probably wont if v1 server is shut down. 

Subha: How much work to support v2?  -- to fred

Frederic: prefer direct integration. wc integration isnt the nicest UI Biggest problem is mobile to mobile. wc does not fix. n

Eric: how do you expect user to use dapp if you dont upgrade? 

Frederic: we have an in app wallet.  Deep links to wallets (valora) would be better than wc. 

Jean: deep link integration (dapp kit) was removed from valora. It didn't work between mobile and web. direct link works well on android as you can choose which wallet but not good on ios as only one app can open specific link

Frederic:  how do we cooporate in crypto world?  are we seemless enough for people to choose to use? not enough scale. 

Eric: what is the ask here Frederic? what is the actionable item? 

Frederic: building something with direct integration with valora. 

Subha: you're saying clabs/ valora should build wc competitor? 

Eric: kinda on right track. integrated wallets are the way to go. Embed functionality with native mobile app. Local signer is safer too. with on device encryption.

**Comment by aaronmgdr** [comment]
# React - Celo Support

Jean: is React celo support staying? 

Dan: Main focus is fee currency. Issue: essentially supporting a tool where lots of others do the same thing. Ideal would be adding fee currency support to existing tools like rainbowkit wagmi. No reason to stop supporting as long as its used. 

It is not going to be removed or support dropped.  just upgrade to WCv2. 

future if it will be heavily influenced by wallets and dapps. 


jean: Had issues upgrading to v5. 

Jean: my app is more in maintainence mode: so want to do minimum work. 

Jean: Rainbowkit probably a bit better on mobile 

Jean: RC heavy on the bundle size> (true facts)

Dan: RC  -> ContatratKit _. Web 3 all big. Ethers much smaller.

**Comment by aaronmgdr** [comment]
# ecosystem trends? 

## id wallet (frederic)
* strong initiative in europe on digital ID 
* at some point regulation pops up. will be forced to have kyc. will need integration between digital ID and crypto. (eventually) 
* should celo integrate some kyc preemptively?
* europe has mica(spelling?) forced to have exchange of digital identity. even P2P 

* eric: seems important but not super pressing
* this seems relevent https://www.centre.io/verite

*  Fred: eu  so offical and game changer. dont believe it will be a private standard. will be national governments issuing digital IDs that create mass adoption. 

* subha: seems less relevent for short term.

## payment rail. 
* Frederic: will celo try to provide a payment rail? 
* Subha: we started this but timing wasnt right, had the downturn. 
*  clabs looking and providing the public infra. 

* (s) how would you want payment rails to be built? 
* (s) what is scope of payment rails
* (f) poly moving into digital ID. Not yet mass adoption to zk yet. what is positioning of celo? would a payment rail fit
* eric: what is payment rail? blockchain is imo
* fred: yes but between whom and whom? crypto is only crypot to crypto. not competitive with national banking. strong believer in Remitance but need CICO. 
* eric: shout out to Valora and FiatConnect. integrations growing. 
* fred: but dont have corridor yet. 
* eric: infra is there. but is it being used. so dont understand questions
* fred: real cash going to shop give cash get money transmited to third world country then cash out. (e2e corridor) 
* fred: need complete solution. 
* eric: sounds like you want more on and off ramp providers. 
* eric: thats probably more a foundation thing. steller + money gram doing probably will be existing tradtitional money transmitters integrating with blockchain

subha what should we build NOW? 

job is clabs is to build infra so someone else can build killer apps. 

Fred: we will jump in. looking for leadership / direction from celo.  remittance, digital id. could be "tornados" (directions)


eric: the tornado for a chain are typically not created by the core devs. its up to apps to do it. account abastraction could be something clabs can build out.. AA could be could inside of dapps. for better UX. 

Fred: Excel was microsoft's tornado. waiting could mean nothing happens. 

Eric: foundation might be better to entity to hear this. as its more Business partnership.
